resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -104,11 +105,14 @@ private[spark] class BasicExecutorFeatureStep(
     // hostname must be no longer than `KUBERNETES_DNSNAME_MAX_LENGTH`(63) characters,
     // so take the last 63 characters of the pod name as the hostname.
     // This preserves uniqueness since the end of name contains executorId
-    val hostname = name.substring(Math.max(0, name.length - KUBERNETES_DNSNAME_MAX_LENGTH))
-      // Remove non-word characters from the start of the hostname
-      .replaceAll("^[^\\w]+", "")
-      // Replace dangerous characters in the remaining string with a safe alternative.
-      .replaceAll("[^\\w-]+", "_")
+    val hostname = if (executorPodNamePrefix.length > KUBERNETES_POD_NAME_PREFIX_MAX_LENGTH) {
+      val mamePrefix = KubernetesConf.getResourceNamePrefix(
```

Comment:
Calling again `getResourceNamePrefix` worries me as this second call will call `uniqueID` again and the two IDs will be different. This makes hard to match files produced/used by one run. Could you please check the resources names too from this point of view, before and after this PR? I mean where the prefix is used like

```
private def dtSecretName: String = s"${kubernetesConf.resourceNamePrefix}-delegation-tokens"

  private def ktSecretName: String = s"${kubernetesConf.resourceNamePrefix}-kerberos-keytab"

  private def hasKerberosConf: Boolean = krb5CMap.isDefined | krb5File.isDefined

  private def newConfigMapName: String = s"${kubernetesConf.resourceNamePrefix}-krb5-file"
```

Reply:
@attilapiros  sorry for the late response.  I checked existed code, seems there are many places which produce the different `uniqueID`:

* `KubernetesDriverConf`
* `KubernetesExecutorConf` 
* `KubernetesClientUtils`
* `DriverServiceFeatureStep`

Do you think we should make `uniqueID` as a global value ? If so, seems it's worth to do this change in a new PR.